src/main/scala/edu/ie3/osmogrid/io/input/InputDataProvider.scala

```diff
@@ -6,37 +6,140 @@
 
 package edu.ie3.osmogrid.io.input
 
-import akka.actor.typed.Behavior
-import akka.actor.typed.ActorRef
-import akka.actor.typed.PostStop
-import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors, StashBuffer}
+import akka.actor.typed.{ActorRef, Behavior}
+import com.acervera.osm4scala.EntityIterator.fromPbf
+import com.acervera.osm4scala.model.{NodeEntity, RelationEntity, WayEntity}
 import edu.ie3.osmogrid.cfg.OsmoGridConfig
-import edu.ie3.osmogrid.guardian.OsmoGridGuardian
+import edu.ie3.osmogrid.model.{OsmoGridModel, PbfFilter}
+import edu.ie3.util.osm.model.OsmEntity.{Node, Relation, Way}
+import org.locationtech.jts.geom.{
+  Coordinate,
+  LinearRing,
+  Point,
+  Polygon,
+  PrecisionModel
+}
+
+import java.io.{FileInputStream, InputStream}
+import java.time.ZonedDateTime
+import java.util.UUID
+import scala.collection.mutable.ListBuffer
+import scala.util.{Failure, Success, Try, Using}
 
 object InputDataProvider {
 
-  sealed trait Request
-  final case class Read()
-      extends Request // todo this read method should contain configuration parameters for the actual source + potential filter options
-  object Terminate extends Request
+  // external requests
+  sealed trait InputRequest
```

Comment:
Throughout the last discussions we more or less agreed on naming the traits `Request` and `Response`.